packages/react-core/src/components/DataList/utils/decorators/wrappable.tsx

```diff
@@ -0,0 +1,14 @@
+import styles from '@patternfly/react-styles/css/components/DataList/data-list';
```

Comment:
It's an interesting approach to applying a set of modifiers, but a bit different from most of our other components (apart from Table I think). 
Typically we just have individual boolean props that map directly to a className.

Here's the Button component for example:
```
className={css(
        styles.button,
        styles.modifiers[variant],
        isBlock && styles.modifiers.block,
        isDisabled && styles.modifiers.disabled,
        isAriaDisabled && styles.modifiers.ariaDisabled,
        isActive && styles.modifiers.active,
        isInline && variant === ButtonVariant.link && styles.modifiers.inline,
        isLoading !== null && styles.modifiers.progress,
        isLoading && styles.modifiers.inProgress,
        isSmall && styles.modifiers.small,
        isLarge && styles.modifiers.displayLg,
        className
      )}
```

But if you wanted to follow this pattern, I would simplify it and just depend on typescript type safety and removing imports etc. something like

And maybe change the prop to `modifiers`?

```
export type ITransform = 'truncate' | 'breakWord' | 'nowrap';
...
modifiers?: ITransform[];
...
className={css(
    modifiers && modifiers.map(key => {
      if (styles.modifiers[key]) return styles.modifiers[key];
      console.warn(`transform ${key} not found`);
    })
)}
```

Don't have strong feelings for one approach or the other. Maybe @tlabaj and @redallen can weigh in as well.

Reply:
I think we should follow the long explicit pattern like we do with button. I find this pattern clever, but we've already committed to adding modifiers the long-winded typesafe way. Also, global exports like `breakword` aren't self explanatory and shouldn't be in our main index.

If we ever have a component with more than 20 modifiers we should reconsider this, though.

Reply:
@redallen I feel that these properties should not be exposed as individual boolean properties on all DataList sub components, ending up with so many exposed properties is not really elegant.

Take a note that the already exposed properties on the Button don't try to affect the same issue, for example isDisabled, isLoading, all affect different concepts of the button, as for disabling events (Disabled) and adding a spinner (isLoading).

On our case though with the text modifiers, they are all referring to the text transforms, and thus should be grouped like these. From what I see in the Table component, these transforms are also not supposed to be used in combinations, which is another indicators that we should have them exposed as a sigle property.

See https://github.com/patternfly/patternfly-react/blob/master/packages/react-table/src/components/Table/TableText.tsx#L27 

I am sure 2 more of these transform will come, because they already exist on the Table component, these are `wrap` && `fitContent`.

Lastly, I understand you concern, but by using the exact same way TableText does it like here https://github.com/patternfly/patternfly-react/blob/master/packages/react-table/src/components/Table/TableText.tsx#L27
I think we will solve the issue with type checking.